### PR TITLE
Remove Candid authentication and RBAC authorization

### PR DESCRIPTION
**Pull request by markylaing**
Removes Candid and RBAC from LXD, the LXD client, the CLI, the documentation, and all related tooling and tests.

### CODE REVIEW

**Comment by markylaing** [comment]
CLA check is failing because there are too many commits (I'll look into fixing this). In the meantime I can see that all of my commits are signed off with: 
```
$ git log main..HEAD --oneline | wc -l
43
$ git log main..HEAD | grep 'Signed-off-by' | wc -l
43
```

File: lxd-migrate/main_migrate.go [hunk 1]
```diff
@@ -160,7 +160,7 @@ func (c *cmdMigrate) askServer() (lxd.InstanceServer, string, error) {
 	type AuthMethod int
 
 	const (
-		authMethodCandid AuthMethod = iota
+		_ AuthMethod = iota
```
Comment (tomponline) [inline lxd-migrate/main_migrate.go:163]: Do we need to keep the old entry at all? Are the IDs used in a DB/config file or map to an API request ID?

**Review by tomponline — APPROVED** [review]

File: lxd-migrate/main_migrate.go (hunk 1 above)
Comment (markylaing) [reply lxd-migrate/main_migrate.go:163]: I don't think so. I left it to keep things consistent. Except now I realise that the integer value picked when running interactively will vary depending on the available authentication methods, so it makes no odds.

I'll remove it.

Comment (markylaing) [reply lxd-migrate/main_migrate.go]: Done.

**Review by tomponline — APPROVED** [review]

**Comment by markylaing** [comment]
@tomponline apologies - I forgot the patch to remove the old config keys from the database. I'll put up a separate PR.

**Pull request merged**